staging/src/k8s.io/component-base/logs/api/v1/options.go

```diff
@@ -50,58 +61,186 @@ func NewOptions() *Options {
 		panic(fmt.Errorf("internal error while setting klog verbosity: %v", err))
 	}
-	if err := loggingFlags.Lookup("vmodule").Value.Set(o.Config.VModule.String()); err != nil {
+	if err := loggingFlags.Lookup("vmodule").Value.Set(c.VModule.String()); err != nil {
 		panic(fmt.Errorf("internal error while setting klog vmodule: %v", err))
 	}
-	klog.StartFlushDaemon(o.Config.FlushFrequency)
+	klog.StartFlushDaemon(c.FlushFrequency)
 	klog.EnableContextualLogging(contextualLoggingEnabled)
 }
+
+// AddFlags adds command line flags for the configuration.
+func (c *LoggingConfiguration) AddFlags(fs *pflag.FlagSet) {
+	// The help text is generated assuming that flags will eventually use
+	// hyphens, even if currently no normalization function is set for the
+	// flag set yet.
+	unsupportedFlags := strings.Join(unsupportedLoggingFlagNames(cliflag.WordSepNormalizeFunc), ", ")
+	formats := fmt.Sprintf(`"%s"`, strings.Join(logRegistry.list(), `", "`))
+	fs.StringVar(&c.Format, "logging-format", c.Format, fmt.Sprintf("Sets the log format. Permitted formats: %s.\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, unsupportedFlags))
+	// No new log formats should be added after generation is of flag options
+	logRegistry.freeze()
+
+	fs.DurationVar(&c.FlushFrequency, LogFlushFreqFlagName, c.FlushFrequency, "Maximum number of seconds between log flushes")
+	fs.VarP(&c.Verbosity, "v", "v", "number for the log level verbosity")
+	fs.Var(&c.VModule, "vmodule", "comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)")
+
+	// JSON options. We only register them if "json" is a valid format. The
+	// config file API however always has them.
+	if _, err := logRegistry.get("json"); err == nil {
+		fs.BoolVar(&c.Options.JSON.SplitStream, "log-json-split-stream", false, "[Experimental] In JSON format, write error messages to stderr and info messages to stdout. The default is to write a single stream to stdout.")
+		fs.Var(&c.Options.JSON.InfoBufferSize, "log-json-info-buffer-size", "[Experimental] In JSON format with split output streams, the info messages can be buffered for a while to increase performance. The default value of zero bytes disables buffering. The size can be specified as number of bytes (512), multiples of 1000 (1K), multiples of 1024 (2Ki), or powers of those (3M, 4G, 5Mi, 6Gi).")
+	}
+}
+
+// SetRecommendedLoggingConfiguration sets the default logging configuration
+// for fields that are unset.
+//
+// Consumers who embed LoggingConfiguration in their own configuration structs
+// may set custom defaults and then should call this function to add the
+// global defaults.
+func (c *LoggingConfiguration) SetRecommendedLoggingConfiguration() {
+	if c.Format == "" {
+		c.Format = "text"
+	}
+	if c.FlushFrequency == 0 {
+		c.FlushFrequency = 5 * time.Second
```

Comment:
you define this as a constant `LogFlushFreqDefault` ? should it be used here?

Reply:
Yes, good catch. Fixed.